R/intervals_support_funs.R

```diff
@@ -0,0 +1,295 @@
+#' @export
+interval_remove_impute <- function(data, target_impute, target_params = NULL, target_groups = NULL, impute_column = NULL, new_rows_after_original = TRUE) {
+  if (missing(data) || missing(target_impute)) {
+    stop("Both 'data' and 'target_impute' must be provided.")
+  }
+  if (!inherits(data, "PKNCAdata") && !is.data.frame(data)) {
+    stop("The 'data' object must be a PKNCAdata object or a data frame.")
+  }
+  if (!is.character(target_impute)) {
+    stop("'target_impute' must be a character string.")
+  }
+  UseMethod("interval_remove_impute")
+}
+
+#' @export
+interval_remove_impute.PKNCAdata <- function(data, target_impute, target_params = NULL, target_groups = NULL, impute_column = NULL, new_rows_after_original = TRUE) {
+  if (is.null(impute_column) && !is.na(data$impute)) {
+    impute_column <- data$impute
+  }
+  data$intervals <- interval_remove_impute(data$intervals, target_impute, target_params, target_groups, impute_column, new_rows_after_original)
+  data
+}
+
+#' @export
+interval_remove_impute.data.frame <- function(data, target_impute, target_params = NULL, target_groups = NULL, impute_column = NULL, new_rows_after_original = TRUE) {
+  
+  # Add an index column to preserve the original order
+  data <- dplyr::mutate(data, index = dplyr::row_number())
+  
+  # Get all parameter column names in the data frame
+  all_param_options <- setdiff(names(get.interval.cols()), c("start", "end"))
+  param_cols <- intersect(names(data), all_param_options)
+  
+  # Handle target_params
+  if (is.null(target_params)) {
+    # Take all parameter columns present with at least one TRUE value
+    target_params <- param_cols[colSums(data[param_cols]) > 0]
+  } else {
+    # Check that all target_params are present in the data frame
+    missing_params <- setdiff(target_params, param_cols)
+    if (length(missing_params) > 0) {
+      stop("The following target_params are not interval columns and/or known PKNCA parameters: ", paste(missing_params, collapse = ", "))
+      target_params <- intersect(target_params, param_cols)
+    }
+  }
+  
+  # Determine the name of the impute column
+  impute_col <- if (!is.null(impute_column)) {
+    if (!impute_column %in% colnames(data)) {
+      stop("The 'data' object does not contain the specified impute column.")
+    }
+    impute_column
+  } else if ("impute" %in% colnames(data)) {
+    "impute"
+  } else {
+    warning("No default impute column identified. No impute methods to remove. If there is an impute column, please specify it in argument 'impute_column'")
+    return(data %>% dplyr::select(-index))
+  }
+  
+  # Identify the targeted intervals based on the groups
+  if (!is.null(target_groups)) {
+    target_intervals <- dplyr::inner_join(data, target_groups, by = names(target_groups))
+  } else {
+    target_intervals <- data
+  }
+  
+  # Identify the targeted intervals based on the impute method and parameters
+  target_intervals <- target_intervals %>%
```

Comment:
Please simplify this code to base R code to avoid the need for more complex dplyr functions like:

```
impute_split <- strsplit(data[[impute_col]], split = "[ ,]+")
impute_subset <- lapply(X = impute_split, FUN = setdiff, target_impute)
impute_final <- vapply(X = impute_subset, FUN = paste, collapse = ",", FUN.VALUE = "")
```

This is probably a useful helper function so that you can use it on the `PKNCAdata` `$impute` string, too.